rust/main/agents/relayer/src/server/environment_variable.rs

```diff
@@ -0,0 +1,188 @@
+use std::env;
+
+use axum::{extract::State, routing, Json, Router};
+use derive_new::new;
+use serde::{Deserialize, Serialize};
+
+const ENVIRONMENT_VARIABLE: &str = "/environment_variable";
+
+#[derive(Clone, Debug, PartialEq, Eq, Deserialize)]
+pub struct SetEnvironmentVariableRequest {
+    name: String,
+    value: Option<String>,
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
+pub struct EnvironmentVariableResponse {
+    name: String,
+    value: Option<String>,
+    message: String,
+}
+
+#[derive(new, Clone)]
+pub struct EnvironmentVariableApi {}
+
+async fn get_environment_variable(
+    State(_): State<EnvironmentVariableApi>,
+    Json(body): Json<SetEnvironmentVariableRequest>,
+) -> Result<Json<EnvironmentVariableResponse>, String> {
+    let value = env::var(&body.name).ok();
+
+    let response = EnvironmentVariableResponse {
+        name: body.name,
+        value,
+        message: "got".to_string(),
+    };
+
+    Ok(Json(response))
+}
+
+async fn set_environment_variable(
+    State(_): State<EnvironmentVariableApi>,
+    Json(body): Json<SetEnvironmentVariableRequest>,
+) -> Result<Json<EnvironmentVariableResponse>, String> {
+    let message = match &body.value {
+        None => {
+            env::remove_var(&body.name);
+            "unset"
+        }
+        Some(value) => {
+            env::set_var(&body.name, value);
+            "set"
+        }
+    };
+
+    let response = EnvironmentVariableResponse {
+        name: body.name,
+        value: body.value,
+        message: message.to_string(),
+    };
+
+    Ok(Json(response))
+}
+
+impl EnvironmentVariableApi {
+    pub fn router(&self) -> Router {
+        Router::new()
+            .route("/", routing::get(get_environment_variable))
```

Comment:
Definitely get what you mean about security concerns 😅

I'd be nervous that someone would publicly expose their 9090 port thinking that it's just metrics, and then they end up exposing some secrets. I'm glad though that this isn't for the validator and relayer only - the worst thing here I guess is just halting liveness or being able to fetch the relayer private key

I do wonder if we should only enable this API if explicitly configured to?

Reply:
I've added a environment variable (sic!) which enables/disables environment variable endpoint.